models/components.go

```diff
@@ -188,24 +188,25 @@ func (c Component) GetHealth() (string, error) {
 			return "", fmt.Errorf("failed to evaluate health expression %s: %v", c.HealthExpr, err)
 		}
 
-		return out, nil
+		return Health(out), nil
 	}
 
-	if h := lo.FromPtr(c.Health); h != "" {
-		return string(h), nil
+	// When HealthExpr is not defined, we take worse of checks, children and the component itself
```

Comment:
This ignores warning status - Should probably make health comparable and then sort it pick the max

Reply:
👍🏽